### DISCUSSION TITLE
Linear elasticity: Dirichlet BC on only one component of the vector field

### DISCUSSION THREAD

**Discussion by karthichockalingam** [mfem/mfem#3158] [Q&A] [answered]
Hello,

I am few to mfem, I would like to apply zero Dirichlet BC on only one component of the displacement (u) vector field. See attached pic, I would like fix u_x = 0 on the left and u_y = 0 on the bottom.

I am looking at ex2 - please let me know how can I modify below lines to accomplish it?

   Array<int> ess_tdof_list, ess_bdr(pmesh->bdr_attributes.Max());
   ess_bdr = 0;
   ess_bdr[0] = 1;
   fespace->GetEssentialTrueDofs(ess_bdr, ess_tdof_list);

Thank you very much!
Karthi.

![Quarter-model-of-infinite-plate-with-hole](https://user-images.githubusercontent.com/39724615/185414513-9a78ddd6-8440-4610-82a9-e33de06bc85a.png)

**Comment by vladotomov** [comment, marked as answer]
He @karthichockalingam,

Apologies that we missed your question. Next time please post in Issues, it's more visible for us this way.

Something like the above BC is done in the Laghos miniapps, see the part of the code [here](https://github.com/CEED/Laghos/blob/2a5de3928f2ac44ef5df342bf70af1ca3f5404f7/laghos.cpp#L418-L432).

Vladimir
